driver/controller.go

```diff
@@ -308,6 +319,10 @@ func (d *Driver) ControllerUnpublishVolume(ctx context.Context, req *csi.Control
 	log.Info("detaching volume")
 	err = d.svc.detachStorage(ctx, req.VolumeId, server.UUID)
 	if err != nil {
+		if errors.Is(err, errUpCloudServerStorageNotFound) {
```

Comment:
Good to check that this error is indeed returned. Atleast by [detaching through our storage API](https://developers.upcloud.com/1.3/9-storages/#request-via-storage-api_1), there is no error returned when storage is not currently attached to anything. Just 200 OK and storage details.

Reply:
@ajmyyra  thanks for the review :+1:  I think that this is what we want. If it's not attached, then we can just return OK. But let's double-check this in PR that addresses the TODO comment.